hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/hdds/scm/storage/TestContainerCommandsEC.java

```diff
@@ -278,6 +292,274 @@ public void testCreateRecoveryContainer() throws Exception {
     }
   }
 
+  private static byte[] getBytesWith(int singleDigitNumber, int total) {
+    StringBuilder builder = new StringBuilder(singleDigitNumber);
+    for (int i = 1; i <= total; i++) {
+      builder.append(singleDigitNumber);
+    }
+    return builder.toString().getBytes(UTF_8);
+  }
+
+  /**
+   * Tests the reconstruction of single missed data index.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes1()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(1));
+  }
+
+  /**
+   * Tests the reconstruction of single missed data index.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes2()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(2));
+  }
+
+  /**
+   * Tests the reconstruction of single missed data index.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes3()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(3));
+  }
+
+  /**
+   * Tests the reconstruction of single missed parity index.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes4()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(4));
+  }
+
+  /**
+   * Tests the reconstruction of single missed parity index.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes5()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(5));
+  }
+
+  /**
+   * Tests the reconstruction of all missed parity indexes.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes45()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(4, 5));
+  }
+
+  /**
+   * Tests the reconstruction of multiple data indexes missed.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes23()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(2, 3));
+  }
+
+  /**
+   * Tests the reconstruction of data and parity missing indexes.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes24()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(2, 4));
+  }
+
+  /**
+   * Tests the reconstruction of data and parity missing indexes.
+   */
+  @Test
+  public void testECReconstructionCoordinatorWithMissingIndexes35()
+      throws Exception {
+    testECReconstructionCoordinator(ImmutableList.of(3, 5));
+  }
```

Comment:
Can be done with single parameterized test:

```suggestion
  @ParameterizedTest
  @MethodSource("recoverableMissingIndexes")
  void testECReconstructionCoordinatorWith(List<Integer> missingIndexes)
      throws Exception {
    testECReconstructionCoordinator(missingIndexes);
  }

  static Stream<List<Integer>> recoverableMissingIndexes() {
    return Stream.concat(
        IntStream.rangeClosed(1, 5).mapToObj(ImmutableList::of),
        Stream.of(
            ImmutableList.of(2, 3),
            ImmutableList.of(2, 4),
            ImmutableList.of(3, 5),
            ImmutableList.of(4, 5)
    );
  }
```

(+ needs imports)

Reply:
Good suggestion. Thanks